### ISSUE TITLE
Error installing LLVM 3.7 on OSX

### ISSUE THREAD

**Issue by gwerbin** [Homebrew/homebrew-versions#1298]
`brew install llvm37 --with-python --with-clang` fails:

```
==>make -C /private/tmp/llvm37-20160622-40035-12w1grx/llvm-3.7.1.src/projects/libcxx install CC=/usr/local/Cellar/llvm37/3.7.1/lib/llvm-3.7/bin/clang CXX=/usr/local/Cellar/llvm3
Last 15 lines from /Users/greg/Library/Logs/Homebrew/llvm37/04.make:
SYMROOT=/private/tmp/llvm37-20160622-40035-12w1grx/llvm-3.7.1.src/projects/libcxx

cd lib && ./buildit
+ for FILE in '../src/*.cpp'
+ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -g -Os -arch i386 -arch x86_64 -nostdinc++ -std=c++11 -fstrict-aliasing -Wall -Wextra -Wshadow -Wconversion -Wpadded -Wstrict-aliasing=2 -Wstrict-overflow=4 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -I../include ../src/algorithm.cpp
In file included from ../src/algorithm.cpp:11:
In file included from ../include/random:1645:
../include/string:1938:44: error: 'basic_string<_CharT, _Traits, _Allocator>' is missing exception specification 'noexcept(is_nothrow_copy_constructible<allocator_type>::value)'
basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
                                           ^
../include/string:1326:40: note: previous declaration is here
    _LIBCPP_INLINE_VISIBILITY explicit basic_string(const allocator_type& __a)
                                       ^
1 error generated.
make: *** [install] Error 1
```

Here is the output from `brew gist-logs llvm37`:
https://gist.github.com/greg-rocketrip/9e3fc86de6a40a2ec1dad2f45b6e871b

**Comment by MikeMcQuaid** [comment]
Remove `pyenv` from your PATH. If not that, it could be one of the other things `brew doctor` warns about.

**Comment by staticfloat** [comment]
I get this as well, but only on OSX 10.11.  OSX 10.9 and 10.10 can build this formula just fine.  This is on a machine with no `brew doctor` warnings, and fully updated. Here's [the `brew --config` output](https://gist.github.com/anonymous/08b69ad2427c95e3c1f1bb60738dbd92).

**Comment by EtherTyper** [comment]
@MikeMcQuaid That's not the issue; I already fixed it with this patch:

https://github.com/ethereum/webthree-umbrella/issues/614#issuecomment-231549573

Hope this helps!
The @EtherTyper

**Comment by MikeMcQuaid** [comment]
@EtherTyper Could you try and open a pull request? [This document](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) should help and I'm happy to walk you through anything else.

Thanks!

**Comment by EtherTyper** [comment]
@MikeMcQuaid Well, I came across Homebrew/legacy-homebrew#31508 from the old brew repo, which says it's purposefully challenging to patch resources. libc++ currently is treated as a resource, so I was wondering if you could should show me how to make it a subformula? Should you just make a subclass extend formula, and then reference it from depends_on like usual?

Thanks,
The @EtherTyper

**Comment by MikeMcQuaid** [comment]
@EtherTyper Oh yeh, I see, sorry. You could consider using an `inreplace` instead of a patch if it's doable that way?

**Comment by EtherTyper** [comment]
@MikeMcQuaid Sure! Should I fix the other LLVM formulae while I'm at it?

**Comment by MikeMcQuaid** [comment]
@EtherTyper If they are broken and you feel like it: please!

**Comment by EtherTyper** [comment]
@MikeMcQuaid Cool! I'm testing the llvm37 fix right now.

**Comment by EtherTyper** [comment]
@MikeMcQuaid The swap didn't work. Can you tell what went wrong?

**Comment by MikeMcQuaid** [comment]
@EtherTyper Needs to be in the `install` block.

**Comment by EtherTyper** [comment]
@MikeMcQuaid Hmm... Tried that, but not sure where the `<libcxx>/include/string` file would be located in the build environment to `inreplace`. I might need some help. 😂

**Comment by MikeMcQuaid** [comment]
@EtherTyper Create a PR and I'll suggest comments inline 😀

**Comment by EtherTyper** [comment]
@MikeMcQuaid #1310

**Comment by EtherTyper** [comment]
Fixes with #1310, http://bot.brew.sh/job/Homebrew%20Versions%20Pull%20Requests/1489/

**Comment by MikeMcQuaid** [comment]
We'll accept PRs for this but we're not actively working on it at this time.
